src/Notepads/Controls/TextEditor/TextEditorCore.cs

```diff
@@ -690,6 +690,20 @@ public bool FindPrevAndSelect(string searchText, bool matchCase, bool matchWhole
             }
         }
 
+        public void SwitchTextFlowDirection(FlowDirection direction)
+        {
+            if (string.IsNullOrEmpty(_content))
+            {
+                // If content is empty, switching text flow direction might not work
+                // Let's not do anything here
+                return;
+            }
+
+            FlowDirection = direction;
+            TextReadingOrder = TextReadingOrder.UseFlowDirection;
+            TextReadingOrder = TextReadingOrder.DetectFromContent;
```

Comment:
You can remove this now since the BiDi issue is mainly caused by `TextScript`-thingy.
The assignment `TextReadingOrder = TextReadingOrder.DetectFromContent` in the constructor could also be removed.

Reply:
Please remove this following my PR. Otherwise changing direction may not work.